pybamm/models/base_model.py

```diff
@@ -274,37 +275,60 @@ def timescale(self, value):
+            self._parameters = self._find_symbols(
+                (pybamm.Parameter, pybamm.InputParameter)
+            )
         return self._parameters
 
-    def _find_parameters(self):
-        """Find all the parameters in the model"""
-        unpacker = pybamm.SymbolUnpacker((pybamm.Parameter, pybamm.InputParameter))
-        all_parameters = unpacker.unpack_list_of_symbols(
-            list(self.rhs.values())
-            + list(self.algebraic.values())
-            + list(self.initial_conditions.values())
-            + list(self.variables.values())
-            + [event.expression for event in self.events]
-        )
-        return list(all_parameters.values())
-
     @property
     def input_parameters(self):
         """Returns all the input parameters in the model"""
         if self._input_parameters is None:
-            self._input_parameters = self._find_input_parameters()
+            self._input_parameters = self._find_symbols(pybamm.InputParameter)
         return self._input_parameters
 
-    def _find_input_parameters(self):
-        """Find all the input parameters in the model"""
-        unpacker = pybamm.SymbolUnpacker(pybamm.InputParameter)
+    def print_parameter_info(self):
+        if self._parameter_info is None:
+            self._parameter_info = ""
+            parameters = self._find_symbols(pybamm.Parameter)
+            for param in parameters:
+                self._parameter_info += f"{param.name} (Parameter)\n"
+            input_parameters = self._find_symbols(pybamm.InputParameter)
+            for input_param in input_parameters:
+                if input_param.domain == []:
+                    self._parameter_info += f"{input_param.name} (InputParameter)\n"
+                else:
+                    self._parameter_info += (
+                        f"{input_param.name} (InputParameter in {input_param.domain})\n"
+                    )
+            function_parameters = self._find_symbols(pybamm.FunctionParameter)
+            for func_param in function_parameters:
+                # don't double count function parameters
+                if func_param.name not in self._parameter_info:
+                    input_names = "'" + "', '".join(func_param.input_names) + "'"
+                    self._parameter_info += (
+                        f"{func_param.name} (FunctionParameter "
+                        f"with input(s) {input_names})\n"
+                    )
+
+        print(self._parameter_info)
+
+    def _find_symbols(self, typ):
+        """Find all the instances of `typ` in the model"""
+        unpacker = pybamm.SymbolUnpacker(typ)
         all_input_parameters = unpacker.unpack_list_of_symbols(
             list(self.rhs.values())
             + list(self.algebraic.values())
             + list(self.initial_conditions.values())
+            + [
+                x[side][0]
+                for x in self.boundary_conditions.values()
+                for side in ["left", "right"]
```

Comment:
I think the issue with the tests is here. Most models give boundary conditions using "left" and "right" as the keys, but some models use "positive tab", "negative tab" or "no tab", so I guess here we need to check for all these keys (and check if they are provided) in the boundary conditions dicts

Reply:
This is a rough code, when added, passes all the tests except 1 - 
```py
keys = self.boundary_conditions.values()
        side_list = []
        flag = True
        for key in keys:
            if isinstance(key, dict):
                if "positive tab" in key.keys():
                    side_list.append("positive tab")
                    flag = False
                if "negative tab" in key.keys():
                    side_list.append("negative tab")
                    flag = False
                if "no tab" in key.keys():
                    side_list.append("no tab")
                    flag = False

        if flag:
            side_list = ["left", "right"]
```
Editing `for side in ["left", "right"]` to `for side in side_list`

The one not passing is `test_plot_2plus1D_spm (test_plotting.test_quick_plot.TestQuickPlot` giving the error `KeyError: 'negative tab'`

Should it be checked like this or is there a better alternative?

Reply:
would this do the same?
```python3
[
                x[side][0]
                for x in model.boundary_conditions.values()
                for side in x.keys()
]
```

Reply:
The tests did pass locally using this, pushed the changes. Thanks!